scheduler/scheduler-node/src/main/java/org/ow2/proactive/scheduler/task/TaskLauncher.java

```diff
@@ -148,12 +144,18 @@ public void doTask(ExecutableContainer executableContainer, TaskResult[] previou
             progressFileReader.start(dataspaces.getScratchFolder(), taskId);
 
             TaskContext context = new TaskContext(executableContainer, initializer, previousTasksResults,
-                dataspaces.getScratchURI(), dataspaces.getInputURI(), dataspaces.getOutputURI(),
-                dataspaces.getUserURI(), dataspaces.getGlobalURI(), progressFileReader.getProgressFile()
-                        .toString(), getHostname());
+                    dataspaces.getScratchURI(), dataspaces.getInputURI(), dataspaces.getOutputURI(),
+                    dataspaces.getUserURI(), dataspaces.getGlobalURI(), progressFileReader.getProgressFile()
+                    .toString(), getHostname());
 
             File workingDir = getTaskWorkingDir(context, dataspaces);
 
+            logger.info("Task working dir : " + workingDir);
```

Comment:
Regarding the time required, have you checked there is no specific implementation of log4j interfaces or related features that would require more work (e.g. custom appender implementations, etc.)? I think it's not just a matter of adapting logging statements but I am maybe wrong...

Besides, IMHO, the upgrade should be done for scheduling and associated projects so that it has a global benefit along with JAR files embedded in a release package.